### PR TITLE
Implement minimal functionality for transport::UTransport

### LINKED ISSUE #113: Implement minimal functionality for transport::UTransport
Just needs the bare minimum in these methods such that work can start on `ZenohUTransport`:

- `UTransport()`
- `send()`
- `registerListener()`
- `getDefaultSource()`
- `cleanupListener()`

The `send()` and `registerListener()` are just very thin wrappers around the pure-virtual versions of the same, `UTransport()` and `getDefaultSource()` only need to set/get the `default_source_` member, and `cleanupListener()` can be empty (by default, the callback connection objects handle disconnect on their own).

This relies on minimal implementations of these utilities:

- [uprotocol::utils::Expected](https://github.com/orgs/eclipse-uprotocol/projects/5/views/2?pane=issue&itemId=63634633) (#111)
- [uprotocol::utils::callbacks](https://github.com/orgs/eclipse-uprotocol/projects/5/views/2?pane=issue&itemId=63634650) (#112)

### PR DESCRIPTION
**Pull request by ruchirchauhan**
Implements minimum functionality for transport::UTransport.
Fixes  #113

### CODE REVIEW

File: src/transport/UTransport.cpp [hunk 1]
```diff
@@ -10,3 +10,61 @@
...
+
+using namespace uprotocol;
+using namespace uprotocol::transport;
+namespace UriValidator = uprotocol::datamodel::validator::uri;
+namespace MessageValidator = uprotocol::datamodel::validator::message;
+
+UTransport::UTransport(const v1::UUri& defaultSrc)
+    : defaultSource_(defaultSrc) {
+
+    auto [bOk, reason] = UriValidator::isValidRpcResponse(defaultSource_);
```
Comment (gregmedd) [inline src/transport/UTransport.cpp:24]: The PR should be rebased to the latest up-cpp main. A new validator has been added for this check to remove ambiguity: `isValidDefaultSource()`

File: src/transport/UTransport.cpp [hunk 2]
```diff
@@ -10,3 +10,61 @@
...
+    if (! bOk) {
```
Comment (gregmedd) [inline src/transport/UTransport.cpp:25]: Minor: a variable name like `sourceIsOk` or `sourceOk` or `sourceGood` would be prefered to the more ambiguous `bOk`:

```suggestion
    auto [sourceOk, reason] = UriValidator::isValidRpcResponse(defaultSource_);
    if (! sourceOk) {
```

File: src/transport/UTransport.cpp [hunk 3]
```diff
@@ -10,3 +10,61 @@
...
 
 #include "up-cpp/transport/UTransport.h"
+#include "up-cpp/datamodel/validator/UUri.h"
+#include "up-cpp/datamodel/validator/UMessage.h"
+
+using namespace uprotocol;
+using namespace uprotocol::transport;
+namespace UriValidator = uprotocol::datamodel::validator::uri;
+namespace MessageValidator = uprotocol::datamodel::validator::message;
+
```
Comment (gregmedd) [inline src/transport/UTransport.cpp:20]: The implementations for `UTransport` should be in the `uprotocol::transport` namespace. `using namespace` is not adequate here.

```suggestion
namespace uprotocol::transport {

namespace UriValidator = datamodel::validator::uri;
namespace MessageValidator = datamodel::validator::message;

```

Being inside the namespace also gives the same benefits of the `using namespace` statements - the current and parent namespaces are automatically available.

**Review by gregmedd — CHANGES_REQUESTED** [review]
A few really minor changes are needed.

This also needs to pass the linter. From the root of the repo, run `lint/clang-format.sh`. It will edit the files in your work tree to comply with the formatting rules.

File: src/transport/UTransport.cpp (hunk 2 above)
Comment (ruchirchauhan) [reply src/transport/UTransport.cpp:25]: done

File: src/transport/UTransport.cpp (hunk 1 above)
Comment (ruchirchauhan) [reply src/transport/UTransport.cpp]: Done

**Review by gregmedd — APPROVED** [review]
Looks good to me

**Pull request merged**